components/camera/ffmpeg/ffmpeg.go

```diff
@@ -102,6 +102,11 @@ func NewFFMPEGCamera(ctx context.Context, conf *Config, logger logging.Logger) (
 
 	ffCam.activeBackgroundWorkers.Add(1)
 	viamutils.ManagedGo(func() {
+		select {
+		case <-cancelableCtx.Done():
```

Comment:
What's calling cancel on this context? I'm only [seeing this call](https://github.com/viamrobotics/rdk/pull/3571/files#diff-69ecbb2a3cf2cd3c3f7e16a096f0a8240a33c292743cfbcdf61b9646086dd9e5R128). `ManagedGo` generally confuses me, but I'm not sure if that lambda with the cancel gets called when the "worker" lambda panics?

Reply:
I think the ffmpeg camera is a [sourceBasedCamera](https://github.com/viamrobotics/rdk/blob/cce2eef54ee94dd84e363db66808b7fb3dfc786d/components/camera/camera.go#L135), so it always rebuilds. The Close gets triggered during the resource rebuild step around [here](https://github.com/viamrobotics/rdk/blob/main/robot/impl/resource_manager.go#L852-L862).